File: templates/java-webmvc/subsystem/post/component/src/main/java/{{PACKAGE_NAME}}/post/PostService.java
package {{PACKAGE_NAME}}.post;

import {{PACKAGE_NAME}}.post.model.Post;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

import java.util.List;

@Service
class PostService implements PostProvided {

  private PostRepository postRepository;

  PostService(PostRepository postRepository) {
    this.postRepository = postRepository;
  }

  List<Post> findAll() {
    return this.postRepository.findAll();
  }

  public List<Post> findByUserId(Integer userId) {
    return this.postRepository.findByUserId(userId);
  }

  Post findById(Integer id) {
    return this.postRepository
      .findById(id)
      .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
  }
}

File: templates/kotlin-webmvc/subsystem/post/component/src/main/kotlin/{{PACKAGE_NAME}}/post/PostService.kt
package {{PACKAGE_NAME}}.post

import {{PACKAGE_NAME}}.post.model.Post
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Service
import org.springframework.web.server.ResponseStatusException

@Service
internal class PostService(
  private val postRepository: PostRepository
): PostProvided {

  fun findAll(): List<Post> =
    this.postRepository.findAll()

  override fun findByUserId(userId: Int): List<Post> =
    this.postRepository.findByUserId(userId)

  fun findById(id: Int): Post =
    this.postRepository.findById(id) ?: throw ResponseStatusException(HttpStatus.NOT_FOUND)

}
